.github/workflows/CI.yaml

```diff
@@ -0,0 +1,92 @@
+name: CI workflow
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'docker/**'
+      - 'graphenex/**'
+  
+  push:
+    branches:
+      - main
+    paths:
+      - 'docker/**'
+      - 'graphenex/**'
+
+
+jobs:
+
+  linter:
+    name: "Linter check"
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+
+      - name: 'flake8 python linter'
+        uses: py-actions/flake8@v2
+
+
+  setup_linux:
+    name: "Linux setup"
+    runs-on: ubuntu-latest
+    needs: linter
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+      
+      - name: update
+        run: |
+          sudo apt-get update
+          sudo apt-get upgrade
+
+      - name: 'install dependencies'
+        run: |
+          sudo apt install python3-pip
```

Comment:
```suggestion
      - name: 'install dependencies'
        run: |
          sudo apt-get update
          sudo apt-get install -y \
            --no-install-recommends \
            --allow-unauthenticated python3-pip
```

Reply:
Can you tell me specific versions of dependencies that you use? I will just make requirements.txt file for faster installation on the runners machines

Reply:
You can check the dependencies and their versions in [pyproject.toml](https://github.com/grapheneX/grapheneX/blob/master/pyproject.toml) -> [tool.poetry.dependencies]

Reply:
> You can check the dependencies and their versions in [pyproject.toml](https://github.com/grapheneX/grapheneX/blob/master/pyproject.toml) -> [tool.poetry.dependencies]

Sorry, I have never used poetry before so didn't know what it can do. I see it uses cloned repository .toml file instead of PyPi, so I will just make that one working in workflow and requirements.txt won't be necessary